File: Cpp/PC.cpp
#include "Processor.cpp"
#include "Disk.cpp"
#include "Ram.cpp"

class PC{
    private:
        // atribut
        int totalPrice;
        processor pros;
        Disk dis;
        ram memo;

    public:
        // konstruktor
        PC(){}

        PC(processor p, Disk d, ram r){
            this->pros = p;
            this->dis = d;
            this->memo = r;
            this->totalPrice = this->pros.getPriceP() + this->dis.getPriceD() + this->memo.getPriceR();
        }

        // setter dan getter
        void setTotalPrice(int totalPrice){ this->totalPrice = totalPrice; }
        int getTotalPrice(){ return this->totalPrice; }

        void setProcessor(processor p){ this->pros = p; }
        processor getProcessor(){ return this->pros; }

        void setDisk(Disk d){ this->dis = d; }
        Disk getDisk(){ return this->dis; }

        void setRam(ram r){ this->memo = r; }
        ram getRam(){ return this->memo; }

        // metode print
        void print(){
            cout << "//////// HASIL ////////" << endl;
            cout << "______ Processor ______" << endl;
            cout << "Merk: " << this->pros.getNameP() << endl;
            cout << "Harga: " << this->pros.getPriceP() << endl;
            cout << endl;

            cout << "______ Disk ______" << endl;
            cout << "Type: " << this->dis.getTypeD() << endl;
            cout << "Kapasitas(TB): " << this->dis.getCapacityD() << endl;
            cout << "Harga: " << this->dis.getPriceD() << endl;
            cout << endl;

            cout << "______ RAM _______" << endl;
            cout << "Kapasitas(GB): " << this->memo.getCapacityR() << endl;
            cout << "Harga: " << this->memo.getPriceR() << endl;
            cout << endl;

            cout << "Total harga(Rp): " << getTotalPrice() << endl;
        }
        
        // destructor
        ~PC(){}
        
};

File: Java/pc.java
public class pc {
    // atribut serta nama objek
    private processor pObj;   
    private disk dObj;   
    private ram rObj;
    private int totalPrice;

    // constructor
    public pc() {
    }

    public pc(processor pObj, disk dObj, ram rObj) {
        this.pObj = pObj;
        this.dObj = dObj;
        this.rObj = rObj;
        this.totalPrice = pObj.getPrice() + dObj.getPrice() + rObj.getPrice();
    }

    // setter dan getter
    public processor getpObj() {
        return pObj;
    }

    public void setpObj(processor pObj) {
        this.pObj = pObj;
    }

    public disk getdObj() {
        return dObj;
    }

    public void setdObj(disk dObj) {
        this.dObj = dObj;
    }

    public ram getrObj() {
        return rObj;
    }

    public void setrObj(ram rObj) {
        this.rObj = rObj;
    }

    public int getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(int totalPrice) {
        this.totalPrice = totalPrice;
    }   

    // untuk menampilkan output
    public void print(){
        System.out.println("\n===== PROCESSOR ======");
        System.out.println("Nama        : " + pObj.getName());
        System.out.println("Harga       : Rp " + pObj.getPrice());
        
        System.out.println("\n===== DISK ======");
        System.out.println("Type        : " + dObj.getType());
        System.out.println("Kapasitas   : " + dObj.getCapacity() + " TB");
        System.out.println("Harga       : Rp " + dObj.getPrice());

        System.out.println("\n===== RAM ======");
        System.out.println("Kapasitas   : " + rObj.getCapacity() + " GB");
        System.out.println("Harga       : Rp " + rObj.getPrice());

    }
    
    
}
